lib/rubocop/cop/rspec/aggregate_examples.rb

```diff
@@ -0,0 +1,377 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Checks if example groups contain more than one aggregateable example.
+      #
+      # @see https://github.com/rubocop-hq/rspec-style-guide#expectations-per-example
+      #
+      # This cop is primarily for reducing the cost of repeated expensive
+      # context initialization.
+      #
+      # @example
+      #
+      #   # bad
+      #   describe do
+      #     specify do
+      #       expect(number).to be_positive
+      #       expect(number).to be_odd
+      #     end
+      #
+      #     it { is_expected.to be_prime }
+      #   end
+      #
+      #   # good
+      #   describe do
+      #     specify do
+      #       expect(number).to be_positive
```

Comment:
even better
```
expect(number).to be_positive
  .and be_odd
  .and be_prime
```

Reply:
Definitely!
I didn't go for this since some of the matchers can't be compounded to a single expectation, e.g.
```
is_expected.to validate_length_of(:about).is_at_most(5000)
  .and ...
...
NoMethodError: undefined method `and' for #<Shoulda::Matchers::ActiveModel::ValidateLengthOfMatcher:0x007fd71d699078>
```

I'm not sure why it is so, probably they were never supposed to be used like that due to side effects.
Pretty sure there are others like those `validate_...`, I don't have a list.

Probably it would be a good idea to add a `CompoundExpectations` cop eventually.